server/storage/sqlite_test.go

```diff
@@ -0,0 +1,27 @@
+// +build !mysqldb
```

Comment:
Unfortunately `sqldb_tests` has some sample config generation that's needed by `tuf_store_test`, which didn't get refactored in this PR.  I plan on doing that in the next PR.

That's why `sqlite3` is still needed because without that refactor, the rethinkdb tests still run the `tuf_store_test`, which depends on stuff in `sqldb_test`.